src/MySql.DataLayer.Core/Connection/MySqlConnectionFactory.cs

```diff
@@ -29,5 +32,15 @@ public async Task<MySqlConnection> GetAsync()
 
             return conn;
         }
+
+        public string GetDatabaseName()
```

Comment:
remove this method and automate the `ChangeDataBaseAsync` in the `GetAsync`.
E.g.:

       public async Task<MySqlConnection> GetAsync(string databaseName = null)
        {
            MySqlConnection conn = new MySqlConnection(_connectionString);

            await conn.OpenAsync();

            if(databaseName != null)
            {
                  await conn.ChangeDataBaseAsync(databaseName);
            }
            else if(_databaseName != null)
            {
                  await conn.ChangeDataBaseAsync(_databaseName);
            }`

            return conn;
        }